### DISCUSSION TITLE
Holes in modeled surfaces -- impacts & ways to resolve?

### DISCUSSION THREAD

**Discussion by divyarajasekharan** [simnibs/simnibs#196] [7.Creating head models] [answered]
Hello! This post touches on some issues that have been previously highlighted, including 1. poor charm segmentation of the GM surface in the case of atrophic brains and 2. small holes appearing in the surface segmentations (GM & skull). Unfortunately, we only acquired T1w images for these participants, so charm is running under sub-optimal conditions. 

As previously documented, running the charm segmentation with default parameters on an atrophic brain has resulted in the GM surface being pushed to the outer CSF boundary: 

![image](https://github.com/simnibs/simnibs/assets/91343323/202620a6-6663-40a4-a30c-1ca7d7403c55)

To resolve this issue, we integrated FreeSurfer segmentations of the GM. This led to visually improved multilabel volume segmentations, but some holes appeared in the extracted GM surface: 

![image](https://github.com/simnibs/simnibs/assets/91343323/1dd557ae-044f-42f8-9018-57561a42f292)

The effect on the resulting volumes is not huge, but still leads to incorrect "indentations" in the cortex: 

![image](https://github.com/simnibs/simnibs/assets/91343323/f9f8a1f1-ba6d-4782-8e39-a36523a91efa)

Are there any obvious ways to fix these issues in the surface extraction step within the existing pipeline? I am able to easily fix these holes using conventional mesh repair tools, like Autodesk Meshmixer, when the charm-generated surfaces are exported from gmsh as STLs. However, I am unsure if these repaired surfaces can be re-inserted into the charm meshing pipeline, just prior to the volumetric meshing / tetrahedral filling step. Any thoughts or advice is greatly appreciated!

**Comment by oulap** [comment]
charm uses a volumetric meshing approach so if you want to add your edits to the mesh, you should edit the file called tissue_labeling_upsampled.nii.gz in the label_prep folder. You can then run the volume meshing step again. For example, you could use the FS surfaces to correct the volume segmentation in the tissue_labeling_upsampled.nii.gz file and then re-run. The surfaces you posted do look a bit strange, so not sure exactly what's going on. There was another user asking specifically about atrophied brains, and his segmentations were improved by changing the parameter settings, see here:

https://github.com/simnibs/simnibs/discussions/131#discussioncomment-5636749

We've made quite some progress making the charm GM look better. Hopefully we can include those changes into a not-too-distant release of SimNIBS.

**Reply by divyarajasekharan** [reply]
Thank you very much for your response! It is very helpful to know that simNIBS is now using a volume-based meshing approach. I did read the referenced discussion-- do you still expect these modified parameters to work in the absence of T2 flair images? 

Great to hear that improvements have been made to the charm GM segmentation! Looking forward to the new release.

**Reply by oulap** [reply, marked as answer]
I don't think it hurts to try those parameters even if you don't have T2s. You can always assess the output visually to see if it improved.

**Reply by divyarajasekharan** [reply]
Confirming that the linked parameters significantly improved the gray matter segmentation & resulting volumetric mesh! Thank you very much!
